platform-core-module/src/main/java/com/softicar/platform/core/module/program/execution/ProgramExecutionRunnable.java

```diff
@@ -91,9 +106,36 @@ private void updateOutputAndTerminatedAt() {
 			execution//
 				.setTerminatedAt(DayTime.now())
 				.setFailed(failed)
-				.setOutput(logBuffer.toString())
+				.setOutput(logOutput.toString())
 				.save();
 			transaction.commit();
 		}
 	}
+
+	private static class TimestampedLogOutput implements ILogOutput {
+
+		private final ILogOutput originalOutput;
+
+		public TimestampedLogOutput(ILogOutput originalOutput) {
+
+			this.originalOutput = Objects.requireNonNull(originalOutput);
+		}
+
+		@Override
+		public void logLine(String line) {
+
+			line = Arrays//
+				.asList(line.split("\n"))
+				.stream()
+				.map(subLine -> "[%s] %s".formatted(DayTime.now(), subLine))
```

Comment:
I wonder if we should add the milliseconds like YYYY-MM-DD HH:MM:SS.MMM? I think this could be handy.

Reply:
ok